### PR TITLE
Fix delay in language service activation caused by cpptools.json downloading.

### PR DESCRIPTION
**Pull request by sean-mcmanus**
Also, changed the default workspaceParsingPriority to "highest" and renamed "normal" to "medium".

### CODE REVIEW

**Comment by sean-mcmanus** [comment]
The "normal" enum change is safe without an engine change because it never checks for "normal", but I'll change the engine references to "normal" later.

File: Extension/src/main.ts [hunk 1]
```diff
@@ -138,27 +138,30 @@ export function activate(context: vscode.ExtensionContext): void {
...
-                    tempCommands.forEach((command) => {
-                        command.dispose();
-                    });
-                    tempCommands = [];
-                    LanguageServer.activate(delayedCommandsToExecute);
-                    delayedCommandsToExecute.forEach((command) => {
-                        vscode.commands.executeCommand(command);
-                    });
-                    delayedCommandsToExecute.clear();
+        util.readFileText(util.getExtensionFilePath("cpptools.json"))
```
Comment (pieandcakes) [inline Extension/src/main.ts:141]: Can you check and make sure the file exists before you read it?

**Review by pieandcakes — APPROVED** [review]
Please make sure you squash when you merge. Thanks!

File: Extension/src/main.ts [hunk 2]
```diff
@@ -138,27 +138,30 @@ export function activate(context: vscode.ExtensionContext): void {
...
+            .then((cpptoolsString) => {
+                processCpptoolsJson(cpptoolsString);
+            })
+            .catch((error) => {
+                // We already log telemetry if cpptools.json fails to download.
```
Comment (bobbrow) [inline Extension/src/main.ts:146]: minor: update comment.  This code path does not download anything anymore.

Comment (sean-mcmanus) [reply Extension/src/main.ts:146]: I don't understand. What code path? You mean the comment is misleading?

File: Extension/src/main.ts (hunk 1 above)
Comment (sean-mcmanus) [reply Extension/src/main.ts:141]: That shouldn't be necessary. The file should always be there unless someone manually deletes it or something else impossible happens and we catch any exceptions in that case too.

Comment (pieandcakes) [reply Extension/src/main.ts:141]: But we shouldn't fail activation if the file isn't there, we should activate with baseline values. why should the file missing make activation stop?

Comment (sean-mcmanus) [reply Extension/src/main.ts:141]: We don't. There's a catch, right?

File: Extension/src/main.ts (hunk 2 above)
Comment (bobbrow) [reply Extension/src/main.ts:146]: I take it back.  I made this comment through the web UI and didn't have the full context.

**Review by bobbrow — APPROVED** [review]

File: Extension/src/main.ts (hunk 1 above)
Comment (sean-mcmanus) [reply Extension/src/main.ts:141]: If Andrew wants to check the file's existence in his refactor pull request, I'd be fine with that.

**Pull request merged**